### ISSUE TITLE
Smarter tab-completion

### ISSUE THREAD

**Issue by BlueberryKing** [HiDeoo/YaTA#6]
**Is your feature request related to a problem? Please describe.**

When you type "Kap" and press tab, it autocompletes to "KAPOW" which is understandable when you consider that completions are sorted alphabetically, but frustrating to me as I'd like to have it autocomplete to "Kappa".

**Describe the solution you'd like**

One solution to this problem would be to allow the user to define what I call "completion biases", basically you would tell the system to prioritize completing to "Kappa" whenever possible and ignore its position in alphabetical sorting. This would mean that if I type "K", "Ka" or "Kap", it would complete to "Kappa" rather than "KAPOW" even though the second "p" in "Kappa" comes after the "O" in KAPOW in the alphabet.

**Describe alternatives you've considered**

- Another solution (albeit more time consuming to implement) would be to tab complete based on the amount of times an emote has been used. So you'd keep track of all of the emotes I've used and then sort completions based on that. Since I've used "Kappa" a lot more often than "KAPOW" it would complete to that. 

- Just have the user get good at typing out emotes.

**Additional context**

If you have "Prioritize usernames when auto-completing" enabled, I think biases should only be active when there is no usernames to complete and we switch to "emote-completion-mode" rather than "username-completion-mode".

**Comment by HiDeoo** [comment]
> "completion biases", basically you would tell the system to prioritize completing to "Kappa" whenever possible and ignore its position in alphabetical sorting.

You tell the system to enable this or you tell the system what to prioritize?

If solution 1, how would the system know what to prioritize? If solution 2, what & how would you give that informations, something like bookmarked emotes? 

> tab complete based on the amount of times an emote has been used

Indeed, that could be a solution.

**Comment by BlueberryKing** [comment]
> You tell the system to enable this or you tell the system what to prioritize?

With solution 1, you would tell the system *what to prioritize*. If you don't have anything prioritized, completions should just be sorted alphabetically (like it is now). 

In solution 2, however, you would choose whether you want the *feature enabled or disabled*. Disabling it would would make completions sorted alphabetically again and possibly also disable the emote usage-counter.

I hope this clears things up.

Also, in terms of implementation, I guess this would be done with some sort of JSON file storing how many times each emote has been used. (Of course, you only have to store the emotes that have been used before and defaulting to 0 if the emote isn't in the list)

**Comment by HiDeoo** [comment]
> you would tell the system what to prioritize

So we're talking about raw strings like `Kappa` > `KAPOW` right?

**Comment by BlueberryKing** [comment]
Actually you would just say something like this:
```
Priorities:
- Kappa
```

So the completions list would be:

```
completions: [
 "Kappa", <-- This is at the top
 "KAPOW",
 "Kasomething"
]
```

instead of 

```
completions: [
 "KAPOW",
 "Kappa",
 "Kasomething"
]
```

**Comment by HiDeoo** [comment]
🤔

**Comment by HiDeoo** [comment]
Fixed in 1a433cb.
